### ISSUE TITLE
Consider hosting MCRs on OME downloads server.

### ISSUE THREAD

**Issue by gusferguson** [imperial-photonics/FLIMfit#205]
Downloading the Windows MCR from the Mathworks web site took 50 minutes. I think it would reduce frustration if the links to the appropriate MCR on the FLIMfit downloads page were to MCRs stored on the OME web server - as Michael does for MTools.

This only should this be much quicker, but also ensure that the correct MCR version was downloaded.

**Comment by seanwarren** [comment]
@gusferguson this is a good idea - @imunro do you know who we need to ask to get the MCR downloads hosted on the OME server? We can change the link over easily once it's up

**Comment by imunro** [comment]
Not too sure who's plate that would land on @gusferguson any thoughts ?

**Comment by hflynn** [comment]
Misunderstanding here - the mtools downloads page links to mathworks just the same. We probably could host it though, I'd need to update the downloads page and script and get the MCR uploaded to our server. I'll look into it.

**Comment by kennethgillen** [comment]
Two immediate questions:

1: is it legal to redistribute the MCR?
2: what are the implications of the intermittent drop-outs of the storage which backs downloads.o.org on any code depending on the existence of the MCR there? i.e. how often will it be pulled / if it's down, what happens.

**Comment by gusferguson** [comment]
The licensing is a bit ambiguous:

MATLAB® Application Deployment Programs 
5.1.1. MATLAB® CompilerTM. Licensee may use and distribute the MCR Libraries for the sole purpose of running Licensed User’s Application generated by the MATLAB Compiler. Licensee may deploy, at no cost, copies of such Applications that incorporate the MCR Libraries and compiled versions of M- les from the licensed Programs required for the Application. No further fees shall be due to MathWorks for such deployment within such Applications, regardless of whether the Application is distributed solely for Licensee’s Internal Operations or to ird Parties. 
Licensee may make copies of the MCR Libraries accessible for deployment on an internal server provided such deployment is solely to its Licensed Users for use with an Application and for Licensee’s Internal Operations only. Licensee agrees to employ the same security measures to protect the MCR Libraries as it uses to safeguard Licensee’s own proprietary intellectual property, but in no event less than reasonable care. No further fees shall be due to MathWorks for such server deployment. 
Licensee is prohibited from making copies of the MCR Libraries accessible to any ird Party separate from a Licensed User’s Application.

**Comment by gusferguson** [comment]
Re frequency - infrequent.

Perhaps we could have a link below:
The MCR also available for download from Mathworks site (very slow).

**Comment by imunro** [comment]
In the light of Kenny's valid points I would suggest we either 
a)  leave this until the next release 
or 
b) forget the idea - is Mathworks site really that slow considering users will only need a new MCR when we change Matlab versions ie once a year-ish?

**Comment by gusferguson** [comment]
Yep it is **really** slow - 50 minutes on the Dundee network - that is a considerable barrier to getting started with FLIMfit I feel.

**Comment by seanwarren** [comment]
Thanks @hflynn - I can also update where the windows installer gets the matlab mcr download from so let me know.

**Comment by imunro** [comment]
@gusferguson Confirmed 40+ mins on IC's network also

**Comment by imunro** [comment]
@seanwarren I would prefer not to make changes for this release at this point

**Comment by hflynn** [comment]
I think we may be at cross-purposes here - the only thing *I* can change is where the MCR download buttons on the http://downloads.openmicroscopy.org/flimfit/ pages serves the file from. If the issue is the Windows installer *within* the FLIMfit windows download, I can't do anything about that.

**Comment by imunro** [comment]
Sean was talking about changing the FLIMfit installer to install from the OME server rather than mathworks

**Comment by gusferguson** [comment]
@hflynn - the URL the Windows installer uses can be changed, but doing that would expose us to the availablity issues Kenny raised.

**Comment by hflynn** [comment]
Ah, ok, sorry, I've got the wrong end of the stick.

**Comment by seanwarren** [comment]
Yup but if we can host the MCR on the OME servers I can update where the installer fetches the file

**Comment by gusferguson** [comment]
Quite why Mathsworks doesn't facilitate the inclusion of the libraries in the application is another question!

**Comment by hflynn** [comment]
We need someone with admin permissions to sort hosting the file for you, neither Gus nor I can do that. Will need to speak to @sbesson on Monday and as Ian says, probably a bit late for this release.

**Comment by sbesson** [comment]
Understanding the legitimate usability issue albeit it is largely independent of our scope, I would agree with @kennethgillen's questions and I propose we defer any decision about mirroring external resources like the MATLAB MCR to our own website to future releases.

Many of the questions raised above (licensing, stable URLs, maintenance, availability etc) and maybe others are critical and we should take the time to discuss them.

**Comment by imunro** [comment]
Agreed. To be clear. Definitely not for this release then.

**Comment by imunro** [comment]
I happened to see a member of the Mathworks team over the weekend who "thought" that there may now be some lightweight versions of the MCR available for download (apparently the default one contains everything matlab uses e.g. CUDA)

**Comment by seanwarren** [comment]
Hi all, 

I've created two links on the flimfit server to the MCR downloads:
http://downloads.flimfit.org/mcr/MCR_R2015b_maci64_installer.zip
http://downloads.flimfit.org/mcr/MCR_R2015b_win64_installer.exe

They operate by HTTP 301 redirects so we can point them wherever we like. At the moment they are pointing to a mirror on google cloud storage but if that becomes a problem in the future we can point them back to the matlab servers. Hopefully this should mean that we can change the hardcoded urls in the next release without worrying that they will break in the future if we change where the MCR is stored or have a problem with a server. 

I've contacted Mathworks to find out whether we are allowed to host the runtime ourselves. 

@imunro - re lightweight MCR; that would be good, let me know if you find them! I can't see anything.
